AsyncDisplayKit/ASControlNode.m

```diff
@@ -254,6 +269,7 @@ - (void)addTarget:(id)target action:(SEL)action forControlEvents:(ASControlNodeE
       [targetActions addObject:NSStringFromSelector(action)];
     });
 
+  _targetAdded = YES;
```

Comment:
@maicki that is a reasonable concern, and there definitely would be some amount of overhead. However, in the code path we care about most, this variable will be nil and so it will be extremely fast.  I think the value of limiting total state and eliminating the possibility of it getting out of sync, for example with the remove target case, is high enough to suggest we should go that route that you initially considered!